components/shared/code/DRV/drv_tps20xx.c

```diff
@@ -0,0 +1,176 @@
+ *          P R I V A T E  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+void drv_tps20xx_private_setICEnabled(drv_tps20xx_channel_E channel, bool enabled);
+bool drv_tps20xx_private_getICFaulted(drv_tps20xx_channel_E channel);
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+HW_StatusTypeDef_E drv_tps20xx_init(void)
+{
+    memset(&drv_tps20xx_data, 0x00U, sizeof(drv_tps20xx_data));
+
+    for (uint8_t channel = 0U; channel < DRV_TPS20XX_CHANNEL_COUNT; channel++)
+    {
+        drv_tps20xx_private_setICEnabled(channel, false);
+        drv_tps20xx_data[channel].state = DRV_TPS20XX_STATE_INIT;
+        drv_tps20xx_data[channel].enable_request = false;
+    }
+
+    for (uint8_t channel = 0U; channel < DRV_TPS20XX_CHANNEL_COUNT; channel++)
+    {
+        drv_tps20xx_data[channel].state = (drv_tps20xx_private_getICFaulted(channel)) ?
+                                            DRV_TPS20XX_STATE_ERROR:
+                                            DRV_TPS20XX_STATE_OFF;
+    }
+
+    return HW_OK;
+}
+
+HW_StatusTypeDef_E drv_tps20xx_deInit(void)
+{
+    for (uint8_t channel = 0U; channel < DRV_TPS20XX_CHANNEL_COUNT; channel++)
+    {
+        drv_tps20xx_setEnabled(channel, false);
+    }
+
+    return HW_OK;
+}
+
+void drv_tps20xx_run100ms(void)
+{
+    for (uint8_t channel = 0U; channel < DRV_TPS20XX_CHANNEL_COUNT; channel++)
+    {
+        switch (drv_tps20xx_data[channel].state)
+        {
+            case DRV_TPS20XX_STATE_OFF:
+                if (drv_tps20xx_data[channel].enable_request)
+                {
+                    drv_tps20xx_data[channel].state = DRV_TPS20XX_STATE_ENABLED;
+                    drv_tps20xx_private_setICEnabled(channel, true);
+                }
+                break;
+            case DRV_TPS20XX_STATE_ENABLED:
+                if (drv_tps20xx_private_getICFaulted(channel))
+                {
+                    drv_tps20xx_data[channel].state = DRV_TPS20XX_STATE_FAULTED;
+                    drv_tps20xx_private_setICEnabled(channel, false);
+                }
+                break;
+            case DRV_TPS20XX_STATE_FAULTED:
+                if (drv_tps20xx_channels[channel].auto_reset)
+                {
+                    drv_tps20xx_private_setICEnabled(channel, true);
+                    drv_tps20xx_data[channel].state = DRV_TPS20XX_STATE_RETRY;
+                }
+                break;
+            case DRV_TPS20XX_STATE_RETRY:
+                if (drv_tps20xx_private_getICFaulted(channel))
+                {
+                    drv_tps20xx_data[channel].state = DRV_TPS20XX_STATE_FAULTED;
+                    drv_tps20xx_private_setICEnabled(channel, false);
+                }
```

Comment:
Presently this retry logic isnt 100% great, there is a 100ms-200ms window between detected failure and another rety when in auto reset mode. This is presently unused but we could look at adding a timer here in the future

Reply:
Added a fixup which incorporates the timer driver and integration into the tps20xx driver